src/psyclone/tests/psyad/main_test.py

```diff
@@ -169,6 +206,38 @@ def test_main_fileout(tmpdir, capsys):
     assert expected in data
 
 
+def test_main_t_option(tmpdir, capsys):
+    ''' Test that the -t option causes the test harness to be generated. '''
+    filename_in = str(tmpdir.join("tl.f90"))
+    filename_out = str(tmpdir.join("ad.f90"))
+    with open(filename_in, "a") as my_file:
+        my_file.write(TEST_MOD)
+    main([filename_in, "-oad", filename_out, "-t"])
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert EXPECTED_HARNESS_CODE in output
+
+
+@pytest.mark.parametrize("extra_args", [[], ["-t"]])
+def test_main_otest_option(tmpdir, capsys, extra_args):
+    ''' Test that the -otest option switches on test-harness generation and
+    causes the result to be written to file. We check with and without the
+    (superfluous) '-t' flag. '''
+    filename_in = str(tmpdir.join("tl.f90"))
+    filename_out = str(tmpdir.join("ad.f90"))
+    harness_out = str(tmpdir.join("harness.f90"))
+    with open(filename_in, "a") as my_file:
+        my_file.write(TEST_MOD)
+    main([filename_in, "-oad", filename_out, "-otest", harness_out] +
+         extra_args)
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert output == ""
+    with open(harness_out, 'r') as my_file:
+        data = my_file.read()
+    assert EXPECTED_HARNESS_CODE in data
+
+
```

Comment:
I think you need to test that the expected logging output is produced - with an xfail on it as caplog does not seem to work in GHA.

Reply:
Done.